### PR TITLE
Example for the UUID type

### PR DESCRIPTION
**Pull request by lholden**
This is an example for the UUID type. :)

### CODE REVIEW

File: examples/uuid/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+#[get("/people/<id>")]
+fn people(id: UUID) -> Result<String, String> {
+    // Because UUID implements the Deref trait, we use Deref coercion to 
+    // convert rocket_contrib::UUID to uuid::Uuid.
+    let person = PEOPLE.get(&id)
+        .ok_or(format!("Person not found for UUID: {}", id))?;
+
+    Ok(format!("We found: {}", person))
```
Comment (SergioBenitez) [inline examples/uuid/src/main.rs]: I think we can roll this into the `PEOPLE.get`call with a `map` before the `ok_or`.

File: examples/uuid/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#![plugin(rocket_codegen)]
+
+#[macro_use]
+extern crate lazy_static;
+extern crate uuid;
+extern crate rocket;
+extern crate rocket_contrib;
+
+use std::collections::HashMap;
+use uuid::Uuid;
```
Comment (SergioBenitez) [inline examples/uuid/src/main.rs]: Why do we need this?

File: examples/uuid/src/main.rs (hunk 1 above)
Comment (lholden) [inline examples/uuid/src/main.rs]: You want this to look something like this?
```rust
Ok(PEOPLE.get(&id)
    .map(|person| format!("We found: {}", person))
    .ok_or(format!("Person not found for UUID: {}", id))?)
```

File: examples/uuid/src/main.rs (hunk 2 above)
Comment (lholden) [inline examples/uuid/src/main.rs]: Common use case is going to be inter-op with the uuid crate. Ex, it's what the Postgres and Diesel crates use for uuid support.

So, it makes sense to actually show how one does this. (Ex, the HashMap is of the uuid::Uuid type and not UUID)

File: contrib/src/uuid.rs [hunk 3]
```diff
@@ -100,6 +101,13 @@ impl PartialEq<uuid_ext::Uuid> for UUID {
     }
 }
 
+impl PartialOrd<uuid_ext::Uuid> for UUID {
```
Comment (SergioBenitez) [inline contrib/src/uuid.rs]: This should be in a separate PR.

File: examples/uuid/src/main.rs (hunk 2 above)
Comment (SergioBenitez) [inline examples/uuid/src/main.rs]: Okay, cool, that makes sense.

**Comment by SergioBenitez** [comment]
Merged in f230d43fd5a7836e263de9977e20fd715ac0502a.
